examples/gno.land/r/demo/keystore/keystore.gno

```diff
@@ -0,0 +1,189 @@
+}
+
+// KeyStore stores the owner-specific avl.Tree
+type KeyStore struct {
+	Owner std.Address
+	Data  avl.Tree
+}
+
+// Set will set a value to a key
+// requires write-access (original caller must be caller)
+func Set(k, v string) string {
+	origOwner := std.GetOrigCaller()
+	return set(origOwner.String(), k, v)
+}
+
+// set (private) will set a key to value
+// requires write-access (original caller must be caller)
+func set(owner, k, v string) string {
+	origOwner := std.GetOrigCaller()
+	if origOwner.String() != owner {
+		return StatusNoWriteAccess
+	}
+	keystoreInterface, exists := data.Get(owner)
+	if !exists {
+		data.Set(owner, &KeyStore{
+			Owner: origOwner,
+			Data:  avl.Tree{},
+		})
+		keystoreInterface, _ = data.Get(owner)
+	}
+	keystore := keystoreInterface.(*KeyStore)
+	keystore.Data.Set(k, v)
+	return StatusOK
+}
+
+// Remove removes a key
+// requires write-access (original owner must be caller)
+func Remove(k string) string {
+	origOwner := std.GetOrigCaller()
+	return remove(origOwner.String(), k)
+}
+
+// remove (private) removes a key
+// requires write-access (original owner must be caller)
+func remove(owner, k string) string {
+	origOwner := std.GetOrigCaller()
+	if origOwner.String() != owner {
+		return StatusNoWriteAccess
+	}
+	keystoreInterface, exists := data.Get(owner)
+	if !exists {
+		data.Set(owner, &KeyStore{
+			Owner: origOwner,
+			Data:  avl.Tree{},
+		})
+		keystoreInterface, _ = data.Get(owner)
+	}
+	keystore := keystoreInterface.(*KeyStore)
+	_, removed := keystore.Data.Remove(k)
+	if !removed {
+		return StatusCouldNotExecute
+	}
+	return StatusOK
+}
+
+// Get returns a value for a key
+// read-only
+func Get(k string) string {
+	origOwner := std.GetOrigCaller()
+	return remove(origOwner.String(), k)
+}
+
+// get (private) returns a value for a key
+// read-only
+func get(owner, k string) string {
+	keystoreInterface, exists := data.Get(owner)
+	if !exists {
+		return StatusNoUser
+	}
+	keystore := keystoreInterface.(*KeyStore)
+	val, found := keystore.Data.Get(k)
+	if !found {
+		return StatusNotFound
+	}
+	return val.(string)
+}
+
+// Size returns size of database
+// read-only
+func Size() string {
+	origOwner := std.GetOrigCaller()
+	return size(origOwner.String())
+}
+
+func size(owner string) string {
+	keystoreInterface, exists := data.Get(owner)
+	if !exists {
+		return StatusNoUser
+	}
+	keystore := keystoreInterface.(*KeyStore)
+	return ufmt.Sprintf("%d", keystore.Data.Size())
+}
+
+// Render provides url access to the functions of the keystore
+// "" -> show all keystores listed by owner
+// "owner" -> show all keys for that owner's keystore
+// "owner:size" -> returns size of owner's keystore
+// "owner:get:key" -> show value for that key in owner's keystore
+// "owner:set:key:value" -> sets a key-value pair for owner's keystore (owner must be caller)
+// "owner:remove:key" -> removes key from owner keystore (owner must be caller)
+func Render(p string) string {
```

Comment:
Why not keep `Render` to be read-only?

Reply:
Agreed, usage of `Render()` should probably be universally understood as read-only for consistency across realm examples though there isn't anything in the docs stating this

Reply:
Thanks. Was not aware that `Render()` should be read-only. Fixed in dd500378